### ISSUE TITLE
NaN lowerbound of quantized log-scale UniformFloatHyperparameter

### ISSUE THREAD

**Issue by baudm** [automl/ConfigSpace#199]
https://github.com/automl/ConfigSpace/blob/22b979c90e07324c07861fd8972ae69b7745c36d/ConfigSpace/hyperparameters.pyx#L469

For any lower bound < `q / 2 + 0.0001`, the said line will result in a negative value for `lower`, resulting in `NaN` for `self._lower`. Can anybody explain the reasoning behind this code?

This causes quantized log-scale variants of `UniformFloatHyperparameter` to be rejected by `ConfigurationSpace` since the default value will always be `NaN`:

```
/home/darwin/configspace_test.py:12: RuntimeWarning: invalid value encountered in log
  uniform_float_hp = CSH.UniformFloatHyperparameter('uni_float_q', lower=1e-4, upper=1e-1, q=1e-5, log=True)
uni_float_q, Type: UniformFloat, Range: [0.0001, 0.1], Default: 0.0031622777, on log-scale, Q: 1e-05
Traceback (most recent call last):
  File "/home/darwin/configspace_test.py", line 14, in <module>
    cs.add_hyperparameter(uniform_float_hp)
  File "ConfigSpace/configuration_space.pyx", line 186, in ConfigSpace.configuration_space.ConfigurationSpace.add_hyperparameter
  File "ConfigSpace/configuration_space.pyx", line 1062, in ConfigSpace.configuration_space.ConfigurationSpace._check_default_configuration
  File "ConfigSpace/configuration_space.pyx", line 1427, in ConfigSpace.configuration_space.Configuration.__init__
  File "ConfigSpace/configuration_space.pyx", line 1457, in ConfigSpace.configuration_space.Configuration.is_valid_configuration
  File "ConfigSpace/c_util.pyx", line 38, in ConfigSpace.c_util.check_configuration
  File "ConfigSpace/c_util.pyx", line 103, in ConfigSpace.c_util.check_configuration
ValueError: Active hyperparameter 'uni_float_q' not specified!```

**Comment by mfeurer** [comment]
Hi, I'm just catching up, is this equivalent to #186?

**Comment by mfeurer** [comment]
Fixed via #202.
